src/main/java/com/takipi/oss/storage/resources/fs/multifetcher/S3Cache.java

```diff
@@ -0,0 +1,8 @@
+package com.takipi.oss.storage.resources.fs.multifetcher;
+
+interface S3Cache {
```

Comment:
As a role I think we shouldn't create an interface if there is only one concrete class. If one day we'll want another implementation we can always extract an interface.

Reply:
There are two implementations of this interface. One is the functioning cache and the other is the dummy cache that does nothing.